woocommerce/woocommerce-gutenberg-products-block
# Add snackbar notices to notices system and implement for coupon ui/ux

Fixes: #1966 (in part)

This finishes off the remainder of #1966.  In this pull:

- I added Snackbar notices for the Notices system. This utilizes the `SnackbarList` component from `@wordpress/components` (aliased in directly).
- I implemented the new snackbar notices with the coupon ui/ux. Snackbar notices are now used for a coupon added successfully, and a coupon removed successfully.

![Screen Recording 2020-03-14 at 09 44 PM](https://user-images.githubusercontent.com/1429108/76693553-880fda80-663d-11ea-97fd-a856257f388b.gif)

Mobile:

![Screen Recording 2020-03-15 at 09 13 AM](https://user-images.githubusercontent.com/1429108/76702079-59c1e780-669d-11ea-93a8-c1994d455187.gif)

## To test:

- verify coupon logic works as expected
- review notice behaviour - any tweaks, changes needed? 
- this should probably be tested across browsers (I tested Chrome).
- Accessibility should be tested (I think this applies a11y speak automatically - "polite").
- test notices on mobile.
